rbi/core/thread.rbi

```diff
@@ -1479,6 +1479,12 @@ end
 # consumer.join
 # ```
 class Thread::Queue < Object
+  # Creates a new queue instance, optionally using the contents of an enumerable for its initial state.
+  # https://ruby-doc.org/core-3.1.0/Thread/Queue.html#method-c-new
+  sig {void}
+  sig {params(enumerable: Enumerable).void}
+  def initialize(enumerable); end
```

Comment:
Rather than using an overloaded signature, can you define `enumerable` as `enumerable=T.unsafe(nil)` and only have one `sig`?

Overloaded sigs are messy and we try to avoid using them unless required.

Reply:
✅ PTAL